### ISSUE TITLE
[Proposal] Policy Team charter and mission review

### ISSUE THREAD

**Issue by rficcaglia** [cncf/tag-security#607]
### Description:

~~The Policy WG was merged into the SAFE WG which then became sig-security.~~

EDIT: See below from @ultrasaurus (summarized here) 

>Before SAFE WG was formed...several folks involved in Kubernetes Policy SIG were interested in forming a CNCF Policy WG. Given the overlap of concerns, SAFE WG agreed to merge efforts and submit a unified proposal for a CNCF WG.

See PR #65 

Separate Policy ~~WG~~ team meetings and projects continued on, which SHOULD have been under under the sig-security governance, but in fact were de facto governed under the Kubernetes Policy WG agenda. 

Note - all this happened before the documented CNCF WG proposal process was well understood and well grooved, see:

https://github.com/cncf/toc/blob/main/workinggroups/README.md#process

Aside: For an example of the process in action this looks like a good example of another WG:

https://github.com/cncf/toc/issues/584

~~Presumably since the CNCF WG process didn't yet exist,~~  [Kubernetes Policy WG had from the beginning a separate Kubernetes](https://docs.google.com/document/d/1ihFfEfgViKlUMbY2NKxaJzBkgHh-Phk5hqKTzK-NEEs/edit#bookmark=id.3lzg2mnwr5s7) TOC PR:

https://github.com/kubernetes/community/pull/1692

Thus there is in the Kubernetes git repo a Kubernetes Policy WG (under sig-auth) and in CNCF sig-security a CNCF Policy Team.

~~So I propose - that the CNCF SIG-security define a charter/mission for a CNCF Policy WG, define whether that would be homed here in CNCF sig-security or standalone, and follow the (now defined) CNCF process.~~

EDIT: Given @ultrasaurus 's extra history I have modified this a bit:

So I propose to clarify the situation as follows:

* Modify #268 or close and create a new one that formalizes the definition and goals of the CNCF sig-security Policy Team
  * Note references to Kubernetes Policy WG projects and repos as external references and "back matter"
  * Clarify any white paper goals should be scoped appropriately to the broader CNCF sig-security goals, not just Kubernetes. This is distinct from the current Kubernetes Policy WG whitepaper which is decidedly and deliberately k8s centric.
* The current members of both the CNCF Policy Team and the Kubernetes Policy WG are welcome to continue on both, or to self-select whether they want to focus on the broader scope (CNCF Policy Team) or the narrower scope (Kubernetes Policy WG)
* The projects and goals of both can be synergistic but this clarifies the governance for both: Policy Team 100% is governed under sig-security and the chairs should decide its course and assign leads if needed.  The Kubernetes Policy WG is itself cleaning up its governance, under sig-auth and the Kubernetes TOC.

### Impact: Describe the impact of the problem. Who will this help?  How will it help them?

There has been confusion on the CNCF SIG-Security meetings about the scope and mission of the Policy ~~WG~~ Team.  Meanwhile there has been a lot of good momentum in the Kubernetes Policy WG creating concrete deliverables and growing the # participants and derivative projects:

* There is now a CRD for PolicyReports, being used by totally independent groups.  
* There is a Policy specific whitepaper effort in progress. 
* There has been alignment effort to NIST OSCAL for policy compliance for Kubernetes.  

all these things are currently very Kubernetes specific.  

The CNCF SIG-security members have expressed both interest in the Policy area, and concern that the Policy Team goals need to be broader than just Kubernetes. 

By (re)defining the Policy ~~WG~~ Team ~~, formally as defined in the CNCF TOC process, and gathering support from the TOC and SIG-Security and the community,~~ and merging some form of #268 we can both clarify the scope and mission, and clarify the relationship ~~(if any)~~ with the ~~now current de-facto~~ Kubernetes Policy WG and its code and other outputs.

### Anti-Goals

Arresting or impeding progress on the CRD or Policy Whitepaper which should continue to serve the interested Kubernetes community is an anti-goal. If CNCF TOC decides a Policy WG is a good thing, either standalone or as a SIG-security project, then we should engage with the Kubernetes-focused community members and sub-projects and discuss how we can support their efforts and increase their velocity (while also doing things beyond Kubernetes).

~~Jumping to solutions without defining the problem space is an anti-goal.~~
EDIT: too vague and hand wavy upon reflection

### Scope: How much effort will this take?

Not yet determined - depends one scope and TOC guidance.

### TO DO
- [X] STAG Sponsor - @achetal01 
- [ ] Project leader(s)
- [ ] TBD

**Comment by rficcaglia** [comment]
@achetal01 @danpopSD @ultrasaurus @pragashj might have either more historical context to add - or forward looking comments to add! thank you in advance!

**Comment by ultrasaurus** [comment]
Thanks for writing up this issue and capturing some of the history!

Some historical context...

Before SAFE WG was formed, CNCF formalized WG process https://github.com/cncf/toc/pull/106 and some SAFE WG members joined with intention of becoming CNCF Security WG (others found the activities useful regardless of affiliation).  In parallel several folks involved in Kubernetes Policy SIG were interested in forming a CNCF Policy WG with broader charter.  Given the overlap of concerns, SAFE WG agreed to merge efforts and submit a unified proposal for a CNCF WG -- the process was stalled for some time as CNCF TOC decided to change WG process, which led to creation of SIGs.

Would be good to update description -- @rficcaglia feel free if you have time before I get to it.

**Comment by ultrasaurus** [comment]
A small step toward explaining the governance / charter of what we called the Policy Team (to avoid using the WG name, which at the time was confusing, and maybe still is, not sure)  https://github.com/cncf/sig-security/pull/268 -- ideally that PR would be reviewed by people active in Policy team meetings and evaluate what makes sense.

I always wondered if it was really sustainable to have K8s Policy SIG be the same group as the people concerned with broader cloud native policy (e.g. would it be welcoming to non-Kubernetes folk?  what if there are folk who want to focus on K8s and not on other cloud native solutions?)  At the time of that PR, I thought better to document current process than to hang things up on theoretical / potential future issues.

**Comment by rficcaglia** [comment]
>  always wondered if it was really sustainable to have K8s Policy SIG be the same group as the people concerned with broader cloud native policy 

My $.02 being one of those people :) It is possible - I hope I have done this - but it absolutely requires OTHER people who bring a strong voice from each perspective. I think it is good to have a link and liaison who can cross pollinate and build collaboration (I have tried to fill that role, for example adding some k8s specific ideas into the CNCF map project and CNCF cloud whitepaper) .  But I think it is indispensable to have strong SMEs who bring a more opinionated position. For example Anca, Jaya and Jim and others from RedHat and IBM have been a collective code and specification dynamo for the k8s policy-wg and our CRD and whitepaper.  Meanwhile Dan and Aradhna and Kapil and others have been looking at the big picture at the CNCF level and trying to articulate policy needs more broadly.  

> (e.g. would it be welcoming to non-Kubernetes folk? what if there are folk who want to focus on K8s and not on other cloud native solutions?) 

I think welcoming is something like the CNCF inclusivity topic - yes - I think we can and have been inclusive bi-directionally. If anyone feels otherwise, please let me or the sig chairs know ASAP! As to whether the attendees would be interested in both, again I think that depends on their personal interests and "day jobs" so to speak.  I can attest that several members of the k8s policy-wg seem to be very k8s security policy as code focused - and even in that domain, the narrower *configuration* sense of policy, not governance or compliance or other policy concerns.  I am sure many attendees coming from the broader CNCF side are less interested in PSPs or rego rules and more interested governance or risk management or threat management, etc.

That's why I think the current reality is actually a good representation of what the community wants - I know that's a bit of a tautology ;)  Those who want to focus narrowly on k8s policy gravitate to the Wed 8am calls. Those who are more interested in broader policy topics gravitate to the wed 10am calls (or the Asia sig calls) and those (like me) who want to be involved in both, attend both!

So I think we can have our cake and define policy as code for it too :P

**Comment by rficcaglia** [comment]
@ultrasaurus just an aside - I think probably "policy" is the wrong term anyway.  I think there might be a benefit to having a "GRC" sig separate from sig-security.  security is often completely unrelated to compliance and vice versa, eg:.  

* Getting your kubernetes cluster FISMA compliant: lots of non-security requirements
* securing your service mesh: I would definitely NOT start with FISMA documentation :)
* running a CIS benchmark scan and reporting results: maybe security (if I actually fix the findings!), maybe compliance (if I don't care about the findings only the report), maybe both (ideally the case)
* etc etc

In the commercial space I think you see this rarefied. No one is selling "policy" per se - they are selling GRC tools or "cloud posture" tools, or "unified management", or compliance automation, or SOAR, or DevSecOps, or threat management, etc.

**Comment by TheFoxAtWork** [comment]
@rficcaglia @achetal01  would you provide an updated issue description so that we can get this added to the related groups and close the issue?

**Comment by anvega** [comment]
It has been a few years since there has been activity on this issue. It's unclear from the strikethrough text and edits to the original post if anything outstanding remains to be addressed. If there is a proposal for a GRC WG or SIG, I suggest creating a new proposal for it if it's still an interest or concern. I'll be closing the issue as it has been inactive for so long.
